solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -140,87 +145,92 @@ private SolrZkClient(
     this.solrClassLoader = solrClassLoader;
-    if (!strat.hasZkCredentialsToAddAutomatically()) {
+    if (zkCredentialsProvider == null) {
       zkCredentialsInjector =
           useDefaultCredsAndACLs
               ? createZkCredentialsInjector()
               : new DefaultZkCredentialsInjector();
-      ZkCredentialsProvider zkCredentialsToAddAutomatically =
+      zkCredentialsProvider =
           useDefaultCredsAndACLs
               ? createZkCredentialsToAddAutomatically()
               : new DefaultZkCredentialsProvider();
-      strat.setZkCredentialsToAddAutomatically(zkCredentialsToAddAutomatically);
+    }
+    if (aclProvider == null) {
+      aclProvider = createACLProvider();
+    }
+    if (chroot != null && aclProvider instanceof SecurityAwareZkACLProvider) {
+      this.aclProvider = ((SecurityAwareZkACLProvider) aclProvider).withChroot(chroot);
+    } else {
+      this.aclProvider = aclProvider;
     }
 
     this.zkClientTimeout = zkClientTimeout;
-    // we must retry at least as long as the session timeout
-    zkCmdExecutor = new ZkCmdExecutor(zkClientTimeout, SolrZkClient.this::isClosed);
-    connManager =
-        new ConnectionManager(
-            "ZooKeeperConnection Watcher:" + zkServerAddress,
-            this,
-            zkServerAddress,
-            strat,
-            onReconnect,
-            beforeReconnect,
-            SolrZkClient.this::isClosed);
 
-    try {
-      strat.connect(
-          zkServerAddress,
-          zkClientTimeout,
-          wrapWatcher(connManager),
-          zooKeeper -> {
-            ZooKeeper oldKeeper = keeper;
-            keeper = zooKeeper;
-            try {
-              closeKeeper(oldKeeper);
-            } finally {
-              if (isClosed) {
-                // we may have been closed
-                closeKeeper(SolrZkClient.this.keeper);
-              }
-            }
-          });
-    } catch (Exception e) {
-      connManager.close();
-      if (keeper != null) {
-        try {
-          keeper.close();
-        } catch (InterruptedException e1) {
-          Thread.currentThread().interrupt();
-        }
-      }
-      throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, e);
+    curatorSafeServiceExecutor =
+        ExecutorUtil.newMDCAwareSingleThreadExecutor(
+            new SolrNamedThreadFactory("curator-safeService"));
+
+    RetryPolicy retryPolicy = new ExponentialBackoffRetry(1000, 3);
+    var clientBuilder =
+        CuratorFrameworkFactory.builder()
+            .ensembleProvider(new FixedEnsembleProvider(zkHost))
+            .namespace(chroot)
+            .sessionTimeoutMs(zkClientTimeout)
+            .connectionTimeoutMs(clientConnectTimeout)
+            .aclProvider(this.aclProvider)
+            .authorization(zkCredentialsProvider.getCredentials())
+            .retryPolicy(retryPolicy)
+            .runSafeService(curatorSafeServiceExecutor);
+
+    client = clientBuilder.build();
```

Comment:
then why declare clientBuilder; just tack on a build() and only need the client var